### PR TITLE
docs(python): Add doc example for `datetime_ranges`

### LINKED ISSUE #13161: Add missing docstring examples
### Description

Some of the methods lack any API examples. The attached CSV shows a list of function urls that do not currently have examples: [has_examples.csv](https://github.com/pola-rs/polars/files/13730668/has_examples.csv)

This is a really good first issue. Some of these methods - such as the I/O methods  - are more advanced but there are plenty of simpler expression and series methods there are good for new contributors. If anyone wants to have a go please reply to this issue saying which group of methods you are working on. **It is best to do small PRs with just a few methods at a time**.  

If anyone is having trouble with the contribution process @ me here (or on discord which I see more easily)

### Link

_No response_

### PR DESCRIPTION
**Pull request by NedJWestern**
See #13161

### CODE REVIEW

**Comment by NedJWestern** [comment]
The test fails due to a data frame column width formatting issue. Suggestions?

File: py-polars/polars/functions/range/datetime_range.py [hunk 1]
```diff
@@ -297,6 +297,26 @@ def datetime_ranges(
...
+    >>> df.select(date_range=pl.datetime_ranges("start", "end"))  # doctest: +SKIP
+    shape: (2, 1)
+    ┌─────────────────────────────────────────────────────────────────┐
+    │ date_range                                                      │
+    │ ---                                                             │
+    │ list[datetime[μs]]                                              │
+    ╞═════════════════════════════════════════════════════════════════╡
+    │ [2022-01-01 00:00:00, 2022-01-02 00:00:00, 2022-01-03 00:00:00] │
+    │ [2022-01-02 00:00:00, 2022-01-03 00:00:00]                      │
+    └─────────────────────────────────────────────────────────────────┘
```
Comment (alexander-beedie) [inline py-polars/polars/functions/range/datetime_range.py:319]:
```suggestion
>>> with pl.Config(fmt_str_lengths=65):
...    print(df.select(date_range=pl.datetime_ranges("start", "end")))
shape: (2, 1)
┌─────────────────────────────────────────────────────────────────┐
│ date_range                                                      │
│ ---                                                             │
│ list[datetime[μs]]                                              │
╞═════════════════════════════════════════════════════════════════╡
│ [2022-01-01 00:00:00, 2022-01-02 00:00:00, 2022-01-03 00:00:00] │
│ [2022-01-02 00:00:00, 2022-01-03 00:00:00]                      │
└─────────────────────────────────────────────────────────────────┘
```

Comment (alexander-beedie) [reply py-polars/polars/functions/range/datetime_range.py:319]: > Suggestions?

Try the above; demonstrates ad-hoc use of `Config` and allows the full width to display 👌

**Comment by mcrumiller** [comment]
Alternatively (if the issue is that the docstring was too wide) you can simply add `# noqa: W505` after the final `"""`.

```python
"""
    This docstring is way too wide but it doesn't matter because we tell ruff to ignore it
""" # noqa: W505

**Comment by NedJWestern** [comment]
> Alternatively (if the issue is that the docstring was too wide) you can simply add `# noqa: W505` after the final `"""`.
> 
> ```python
> """
>     This docstring is way too wide but it doesn't matter because we tell ruff to ignore this type of error
> """ # noqa: W505
> ```

Ah, that's what I was looking for!

File: py-polars/polars/functions/range/datetime_range.py (hunk 1 above)
Comment (NedJWestern) [reply py-polars/polars/functions/range/datetime_range.py]: OK, cool. Thanks!

**Comment by NedJWestern** [comment]
Apologies for the false starts, I need to get the doctests running locally.

**Review by stinodego — APPROVED** [review]

**Pull request merged**